service/src/main/java/tds/assessment/repositories/impl/ItemQueryRepositoryImpl.java

```diff
@@ -0,0 +1,133 @@
+                        "    A.groupkey,\n" +
+                        "    A.itemposition AS position,\n" +
+                        "    A.isactive,\n" +
+                        "    A.isfieldtest,\n" +
+                        "    A.isrequired, \n" +
+                        "    A.strandname \n" +
+                        "FROM \n" +
+                        "    itembank.tblsetofadminitems as A \n" +
+                        "JOIN \n" +
+                        "    itembank.tblitem I \n" +
+                        "    ON I._key = A._fk_item \n" +
+                        "JOIN itembank.tblsetofadminsubjects segments \n" +
+                        "    ON segments._key = A._fk_adminsubject \n" +
+                        "WHERE \n" +
+                        "   segments.virtualtest = :key OR segments._key = :key";
+
+        List<Item> items = jdbcTemplate.query(itemsSQL, parameters, (RowMapper<Item>) (rs, row) -> {
+                    Item item = new Item(rs.getString("id"));
+                    item.setSegmentKey(rs.getString("segmentKey"));
+                    item.setItemType(rs.getString("itemtype"));
+                    item.setGroupId(rs.getString("groupid"));
+                    item.setGroupKey(rs.getString("groupkey"));
+                    item.setPosition(rs.getInt("position"));
+                    item.setActive(rs.getBoolean("isactive"));
+                    item.setFieldTest(rs.getBoolean("isfieldtest"));
+                    item.setRequired(rs.getBoolean("isrequired"));
+                    item.setStrand(rs.getString("strandname"));
+                    return item;
+                }
+        );
+
+
+
+        return items;
+    }
+
+    @Override
+    public List<ItemProperty> findActiveItemsProperties(final String assessmentKey) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("key", assessmentKey);
+
+        final String SQL =
+            "SELECT \n" +
+            "   _fk_item AS itemId, \n" +
+            "   propname AS name, \n" +
+            "   propvalue AS value, \n" +
+            "   propdescription AS description \n" +
+            "FROM \n" +
+            "   itembank.tblitemprops P \n" +
+            "JOIN itembank.tblsetofadminsubjects segments \n" +
+            "    ON segments._key = P._fk_adminsubject \n" +
+            "WHERE \n" +
+            "   segments.virtualtest = :key OR segments._key = :key AND" +
+            "   isActive = 1";
+
+        return jdbcTemplate.query(SQL, parameters, (rs, row) ->
+            new ItemProperty(
+                rs.getString("name"),
+                rs.getString("value"),
+                rs.getString("description"),
+                rs.getString("itemId")
+            )
+        );
+    }
+
+    @Override
+    public List<ItemConstraint> findItemConstraintsForAssessment(final String assessmentId) {
+        Map<String, Object> parameters = new HashMap<>();
+//        parameters.put("clientName", clientName);
```

Comment:
I thought we left with a pretty clear understanding of how it would work after standup.

`assessments/{clientName}/{assessmentKey}`

The reason I asked for it to not be included is because the URL change results in changes across other projects stopping this from being merged until this work is complete.  Branch off this branch if you want.  I see no reason it all needs to be in this pull.